src/NuGet.Clients/NuGet.Tools/NuGetPackage.cs

```diff
@@ -948,37 +985,66 @@ private async Task<IVsWindowFrame> CreateDocWindowForSolutionAsync()
+        {
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            if (ShouldMEFBeInitialized(isOpenedByURI))
+            {
+                await InitializeMEFAsync();
+            }
+
+            var windowFrame = await FindExistingSolutionWindowFrameAsync();
+            if (windowFrame == null)
+            {
+                //Create the window frame
+                if (isOpenedByURI && !await SolutionManager.Value.IsSolutionAvailableAsync())
                 {
-                    await InitializeMEFAsync();
-                }
+                    EventHandler handler = null;
+                    handler = (s, e) =>
+                    {
+                        SolutionManager.Value.SolutionOpened -= handler;
 
-                var windowFrame = await FindExistingSolutionWindowFrameAsync();
-                if (windowFrame == null)
+                        NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                        {
+                            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                            windowFrame = await CreateDocWindowForSolutionAsync();
+                            DisplayPMUISearch(windowFrame, searchText, isOpenedByURI, version);
+                        }).PostOnFailure(nameof(NuGetPackage), nameof(ShowManageLibraryPackageDialogAsync));
+                    };
+                    SolutionManager.Value.SolutionOpened += handler;
+                    return;
+                }
+                else
                 {
-                    // Create the window frame
                     windowFrame = await CreateDocWindowForSolutionAsync();
                 }
+            }
+            DisplayPMUISearch(windowFrame, searchText, isOpenedByURI, version);
+        }
 
-                if (windowFrame != null)
-                {
-                    // process search string
-                    string parameterString = null;
-                    var args = e as OleMenuCmdEventArgs;
-                    if (args != null)
-                    {
-                        parameterString = args.InValue as string;
-                    }
-                    var searchText = GetSearchText(parameterString);
-                    Search(windowFrame, searchText);
+        private void DisplayPMUISearch(IVsWindowFrame windowFrame, string searchText, bool isOpenedByURI = false, string version = null)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+            if (windowFrame != null)
+            {
+                Search(windowFrame, searchText);
 
-                    windowFrame.Show();
+                if (isOpenedByURI)
+                {
+                    var packageManagerControl = VsUtility.GetPackageManagerControl(windowFrame);
+                    packageManagerControl?.SelectFirstPackage(version);
```

Comment:
This can just be called in the one scenario where the package is opened by a URI rather than having another check for isOpenedByURI in this method.